src/parser.rs

```diff
@@ -3,9 +3,64 @@ use error::Error;
 
 use nom::types::CompleteStr;
 use nom::{eol, space, space0};
+use nom_locate::LocatedSpan;
 
+use std::fmt;
 use std::num;
 
+/// Captures the offset, line number and remaining text for the current position of the parser.
+pub type Span<'a> = LocatedSpan<CompleteStr<'a>>;
+
+/// Parser error codes.
+///
+/// We use a nom's simple error flavour, which only allows to return error codes
+/// when parser fails.
+pub enum ErrorCode {
+    ExpectedStringExpression,
+    Unknown,
+}
+
+impl From<u32> for ErrorCode {
+    fn from(code: u32) -> Self {
```

Comment:
So the error code is coming from nom. What do you think about using [num-dervie](https://docs.rs/num-derive/0.2.2/num_derive/) to avoid any source or error here?